lib/singlepass-backend/src/codegen_x64.rs

```diff
@@ -6339,6 +6833,14 @@ impl FunctionCodeGenerator<CodegenError> for X64FunctionCode {
 
```

Comment:
Don't you need to pass in canonicalized nans? If the value is cncl_f32, it needs to be canonicalized because the callee might observe it?